src/server/scripts/Northrend/zone_sholazar_basin.cpp

```diff
@@ -50,109 +52,129 @@ enum EngineerHelice
-                case 0:
-                    Talk(SAY_WP_2);
-                    break;
-                case 1:
-                    Talk(SAY_WP_3);
-                    me->CastSpell({ 5918.33f, 5372.91f, -98.770f }, SPELL_EXPLODE_CRYSTAL, true);
-                    me->SummonGameObject(184743, 5918.33f, 5372.91f, -98.770f, 0, QuaternionData(), 3s, GO_SUMMON_TIMED_DESPAWN);
-                    me->HandleEmoteCommand(EMOTE_ONESHOT_LAUGH);
-                    break;
-                case 2:
-                    Talk(SAY_WP_4);
-                    break;
-                case 7:
-                    Talk(SAY_WP_5);
-                    break;
-                case 8:
-                    me->CastSpell({ 5887.37f, 5379.39f, -91.289f }, SPELL_EXPLODE_CRYSTAL, true);
-                    me->SummonGameObject(184743, 5887.37f, 5379.39f, -91.289f, 0, QuaternionData(), 3s, GO_SUMMON_TIMED_DESPAWN);
-                    me->HandleEmoteCommand(EMOTE_ONESHOT_LAUGH);
-                    break;
-                case 9:
-                    Talk(SAY_WP_6);
-                    break;
-                case 13:
-                    if (player)
-                    {
-                        player->GroupEventHappens(QUEST_DISASTER, me);
-                        Talk(SAY_WP_7);
-                    }
-                    break;
-            }
+            Start(false, false, player->GetGUID());
+            Talk(SAY_WP_1);
         }
+    }
+};
 
-        void Reset() override
-        {
-            Initialize();
-        }
+class spell_q12688_detonate_1 : public SpellScript
+{
+    PrepareSpellScript(spell_q12688_detonate_1);
 
-        void JustDied(Unit* /*killer*/) override
-        {
-            if (HasEscortState(STATE_ESCORT_ESCORTING))
-            {
-                if (Player* player = GetPlayerForEscort())
-                    player->FailQuest(QUEST_DISASTER);
-            }
-        }
+    static constexpr uint32 SPAWN_GROUP_FLAMES = 67;
 
-        void UpdateAI(uint32 uiDiff) override
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_EXPLOSION });
+    }
+
+    void HandleDummyEffect(SpellEffIndex /*effIndex*/)
+    {
+        if (Unit* target = GetHitUnit())
         {
-            EscortAI::UpdateAI(uiDiff);
+            target->CastSpell(target, SPELL_EXPLOSION);
 
-            if (HasEscortState(STATE_ESCORT_ESCORTING))
-            {
-                if (m_uiChatTimer <= uiDiff)
-                {
-                    m_uiChatTimer = 12000;
-                }
-                else
-                    m_uiChatTimer -= uiDiff;
-            }
+            std::vector<WorldObject*> flames;
+            target->GetMap()->SpawnGroupSpawn(SPAWN_GROUP_FLAMES, false, false, &flames);
+            target->GetMap()->SetSpawnGroupInactive(SPAWN_GROUP_FLAMES);
+            for (WorldObject* flame : flames)
+                dynamic_cast<GameObject*>(flame)->DespawnOrUnsummon(20s);
```

Comment:
@Carbenium coverity reported this as static analysis issue as dynamic_cast can return null. Either null-check ToGameObject() result or assert (null check ToGameObject() should be better as there is little gain on crashing the whole server with an assert)

I can do it in the next days if you are busy.

Coverity issues 1457830 and 1457831 for both dynamic_cast lines in this file (line 172 too)